Ficek_QuantiApp/Ficek_QuantiApp/Domain/Tests/RocketClientTests/RocketClient+happyTests.swift

```diff
@@ -3,78 +3,112 @@ import XCTest
+    var valueRecieved = false
+    let expectation = expectation(description: "Awaiting Success")
     var cancellables = Set<AnyCancellable>()
+    let mockResponse = HTTPURLResponse(
+      url: URL(string: "https://api.spacexdata.com/v4/rockets")!,
+      statusCode: 200,
+      httpVersion: nil,
+      headerFields: [
+        "\(HTTPHeaderName.acceptType.rawValue)": "\(AcceptTypeValue.json.rawValue)",
+        "\(HTTPHeaderName.contentType.rawValue)": "\(ContentTypeValue.json.rawValue)"
+      ]
+    )
     
-    var testScheduler: TestScheduler<
-        DispatchQueue.SchedulerTimeType,
-        DispatchQueue.SchedulerOptions
-    >! = DispatchQueue.test
+    let mockUrlRequester = URLRequester { _ in
+      Just((successResponse, mockResponse!))
+        .setFailureType(to: URLError.self)
+        .eraseToAnyPublisher()
+    }
     
-    override func invokeTest() {
-        withDependencies {
-            $0.heightConverter = .live()
-            $0.enginesConverter = .live()
-            $0.secondStageConverter = .live()
-            $0.firstStageConverter = .live()
-            $0.massConverter = .live()
-            $0.diameterConverter = .live()
-            $0.rocketConverter = .live()
-            $0.rocketsConverter = .live()
-        } operation: {
-          super.invokeTest()
-        }
-      }
+    let networkClient = NetworkClient(
+      urlRequester: mockUrlRequester,
+      networkMonitorClient: .live(onQueue: .main)
+    )
     
-    func test_fetchAllRockets_sucessful() throws {
-        let successResponse = try JSONEncoder().encode([RocketDTO].mock)
-        let mockResponse = HTTPURLResponse(
-              url: URL(string: "https://api.spacexdata.com/v4/rockets")!,
-              statusCode: 200,
-              httpVersion: nil,
-              headerFields: [
-                "\(HTTPHeaderName.acceptType.rawValue)": "\(AcceptTypeValue.json.rawValue)",
-                "\(HTTPHeaderName.contentType.rawValue)": "\(ContentTypeValue.json.rawValue)"
-              ]
-        )
-        
-        let mockUrlRequester = URLRequester { _ in
-              Just((successResponse, mockResponse!))
-                .setFailureType(to: URLError.self)
-                .eraseToAnyPublisher()
-        }
-        
-        let sut = withDependencies { dependency in
-            let networkClient = NetworkClient(
-                urlRequester: mockUrlRequester,
-                networkMonitorClient: .live(onQueue: .main)
-            )
-            
-            dependency.networkClient = networkClient
-        } operation: {
-            RocketClient.liveValue
+    let sut = withDependencies { dependency in
+      dependency.networkClient = networkClient
+    } operation: {
+      RocketClient.liveValue
+    }
+    
+    sut.fetchAllRocketsCombine()
+      .sink(
+        receiveCompletion: { completion in
+          switch completion {
+          case .finished:
+            expectation.fulfill()
+            valueRecieved = true
```

Comment:
Tak 3 veci k tomuhle:
- `expectation.fullfill` musi bejt az podtim `valueReceived`. Co kdyby se ti naplnila dole ta expectation a zacal se vyhodnocovat ten `XCTAssertTrue` driv nez se `valueReceived` preklopi na `true`?
- tohle nema bejt prece tady, zamysli se nadtim. Kdyz ti neprijde zadna `value`, pouze `finished` stav, tak se vyhodnoti ten test jako passing, pritom to neni ocekavany chovani
- mas tam typo: `valueRecieved` -> `valueReceived`